typescript-view-swapping/main.ts

```diff
@@ -0,0 +1,29 @@
+const $TabContainer = document.querySelector('.tab-container');
+const $AllTab = document.querySelectorAll('.tab');
+const $AllView = document.querySelectorAll('.view');
+
+if (!$TabContainer) throw new Error("Didn't work:(");
+
+function handleClick(event: Event): void {
+  const $eventTarget = event.target as HTMLDivElement;
+
+  for (let i = 0; i < $AllTab.length; i++) {
+    if ($eventTarget.textContent === $AllTab[i].textContent) {
```

Comment:
Missing something right here from instruction #2. We're not ensuring that what we clicked on was a `.tab`. Theoretically we could click anywhere without the `$tabContainer` and this would attempt to run our loops which could cause errors since our `event.target` may not have the properties we are using to check.